code/obj/submachine/seed.dm

```diff
@@ -1084,26 +1084,110 @@
 	attack_ai(mob/user as mob)
 		return src.Attackhand(user)
 
-	attack_hand(var/mob/user as mob)
-		src.add_dialog(user)
-		var/dat = "<B>[src.name]</B><BR><HR>"
-		dat += "<b>Amount to Vend</b>: <A href='?src=\ref[src];amount=1'>[src.vendamt]</A><br>"
-		if (src.category)
-			dat += "<b>Filter</b>: [src.category] <A href='?src=\ref[src];category=1'>(Clear)</A><br>"
-		else
-			dat += "<b>Filter</b>: <A href='?src=\ref[src];category=1'>(Set)</A><br>"
-		if (!src.can_vend)
-			dat+= "<u>Unit currently out of charge. Please wait.</u><br>"
-		dat += "<br>"
+	proc/getPlantImg(var/datum/plant/plant)
+		if(!istype(plant))
+			return
+		var/path = plant.type
+		var/img = plant_base64_cache[path]
+		if(!img)
+			var/icon/result_icon
+			if(plant.crop)
+				var/atom/crop = plant.crop
+				result_icon = icon(initial(crop.icon), initial(crop.icon_state), frame=1)
+			else if(plant.plant_icon)
+				if(plant.sprite)
+					result_icon = icon(plant.plant_icon, "[plant.sprite]-G4", frame=1)
```

Comment:
Slight hesitation from me around building the string for the sprite's icon state to use up here, rather than making this into a proc of some kind (and replacing usages elsewhere with the same proc).

As issues arising should be non-breaking (assuming the change stated down in the tgui files around `null` values is done) and also decently obvious I think it'll be fine for now, not blocking this PR assuming noted null handling is done.

Reply:
> Slight hesitation from me around building the string for the sprite's icon state to use up here, rather than making this into a proc of some kind (and replacing usages elsewhere with the same proc).

Added `datum/plant/proc/getIconState`.